src/pip/_internal/commands/__init__.py

```diff
@@ -3,57 +3,117 @@
 """
 from __future__ import absolute_import
 
-from pip._internal.commands.completion import CompletionCommand
-from pip._internal.commands.configuration import ConfigurationCommand
-from pip._internal.commands.debug import DebugCommand
-from pip._internal.commands.download import DownloadCommand
-from pip._internal.commands.freeze import FreezeCommand
-from pip._internal.commands.hash import HashCommand
-from pip._internal.commands.help import HelpCommand
-from pip._internal.commands.list import ListCommand
-from pip._internal.commands.check import CheckCommand
-from pip._internal.commands.search import SearchCommand
-from pip._internal.commands.show import ShowCommand
-from pip._internal.commands.install import InstallCommand
-from pip._internal.commands.uninstall import UninstallCommand
-from pip._internal.commands.wheel import WheelCommand
+import importlib
 
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
-    from typing import List, Type
+    from typing import Any, Dict, Iterable, List, Tuple
     from pip._internal.cli.base_command import Command
 
+
+commands_dict = {
+    'check': (
+        'pip._internal.commands.check', 'CheckCommand',
```

Comment:
We can probably drop the "pip._internal.commands." prefix and attach it at the point where it's used. It's not a significant compute cost and helps reduce duplication (and chances of typos).

Reply:
That's actually how I had it originally until I encountered that test case you also discovered further on. :)

Reply:
One possible downside  of not fully-qualifying the name is that automatic refactoring tools will not pick it up, but a quick test with PyCharm (CE 2019.1.3) indicates that even with the string as-is it does not recognize this string when I refactor the module `check` to `check2`.